packages/design-system/src/components/Field/Field.tsx

```diff
@@ -253,11 +240,13 @@ const InputWrapper = styled<FlexComponent>(Flex)<{
         return css`
           padding-inline-start: ${props.$hasLeftAction ? props.theme.spaces[3] : props.theme.spaces[4]};
           padding-inline-end: ${props.$hasRightAction ? props.theme.spaces[3] : props.theme.spaces[4]};
+          padding-block: ${props.theme.spaces[1]};
         `;
       default:
         return css`
           padding-inline-start: ${props.$hasLeftAction ? props.theme.spaces[3] : props.theme.spaces[4]};
           padding-inline-end: ${props.$hasRightAction ? props.theme.spaces[3] : props.theme.spaces[4]};
+          padding-block: ${props.theme.spaces[2]};
```

Comment:
Putting the padding outside of the input element creates a problem, as not the visual space dedicated to the input can focus it, which is an accessibility issue:

![CleanShot 2024-07-31 at 11 36 23@2x](https://github.com/user-attachments/assets/0a69ad63-0207-42f6-93ec-24b329428da7)

In terms of aesthetics it also doesn't look great with the autofill highlight from password managers:
![CleanShot 2024-07-31 at 11 29 11@2x](https://github.com/user-attachments/assets/4ec04c50-2a9e-4257-ba51-a4d4255a6ff1)

Reply:
In my opinion the way to get a11y right is to put all the padding on the input element.

I think we should be able to render the endActions using position absolute on top of the input, and the reserve the space underneath by adapting the input's paddingRight based on the presence or absence of an endAction

Reply:
I am not sure to have `absolute` position here as `endAction` could be anything and not just an icon

Reply:
For the paddingRight value, I think you can compute it, by measuring the width of the endAction using a ref, and adding a small gap so the content can't be stuck to it

Then you can add a div wrapper around endAction, set set this div to absolute, and `top: 50%; transformY(-50%)` to ensure vertical centering regardless of the size

Reply:
right dynamically 👍